### PR TITLE
Add `Output dependency tree` by pipdeptree to Actions

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Visualize dependency tree.

This will help editing `pyproject.toml` and debagging errors triggered by dependency changes such as https://github.com/optuna/optuna/actions/runs/4742990938/jobs/8421961600.

## Description of the changes
Add `Output dependency tree`  job after `Output installed packages` to all workflows, which installs `pipdeptree` and run it.

### CODE REVIEW

**Comment by Alnusjaponica** [comment]
Note that `pipdeptree` does not introduce additional dependency (for now).

**Comment by toshihikoyanase** [comment]
The CI failure will be resolved by https://github.com/optuna/optuna/pull/4625. Let me assign the reviewers after that.

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -34,6 +34,10 @@ jobs:
     - name: Output installed packages
       run: |
         pip freeze --all
+    - name: Output dependency tree
+      run: |
+        pip install pipdeptree
+        pipdeptree
```
Comment (toshihikoyanase) [inline .github/workflows/checks.yml:40]: [Notes] Good catch! It's easy to check dependencies on grandchildren. image

**Comment by toshihikoyanase** [comment]
@not522 @cross32768 You may be interested in this PR. Could you join the review, please?

**Review by not522 — APPROVED** [review]
LGTM!

**Review by cross32768 — APPROVED** [review]
LGTM!

**Pull request merged**